### ISSUE TITLE
Is there anything that can be added for `Missing property's type declaration`

### ISSUE THREAD

**Issue by chrisan** [barryvdh/laravel-ide-helper#1034]
Steps to observe behavior:

php 7.4, laravel 6.2, ide helper 2.7

1. Create a model
2. add `protected $guarded = [];`
3. Notice warning: `Missing property's type declaration`
4. add `array` type
5. Notice error: `Type must not be defined (as in base class '\Illuminate\Database\Eloquent\Concerns\GuardsAttributes') `

**Comment by mfn** [comment]
I'm not sure I understand. Can you please provide:
- the command you run
- the full error (stacktrace?) output you get?

thanks

**Comment by chrisan** [comment]
For step one it’s just vanilla Laravel model creation 

‘php artisan make:model Food

There is no error or stack trace.  Sorry if that was misleading, I’m just talking about phpstorm code hints/guides/info.  No PHP is being run

Maybe PhpStorm calls it something different but at step 3 you get a yellow squiggly line and at step 5 you get a red squiggly

That make sense?  I can get a screenshot of the ide when I return home if that would help

**Comment by DannyvdSluijs** [comment]
This seems you PHPstorm is setup to recommend type hints (Perhaps you have PHP inspections installed?)
However Laravel isn’t using the type hints that much therefor you get the error mentioned in step 5. 

Out of experience I know Taylor Otwell doesn’t want to add the typeringen to Laravel. Don’t ask me about the why. So if any bug I would find which plugin or part of PHPstorm is doing the suggestion to add array type hint as that should introspect the extending class. 

Long story short I don’t think this is an issue with ide-helper. As it doesn’t make the ide suggestions.

**Comment by mfn** [comment]
Thanks, this has nothing to do with ide-helper.

Basically as @DannyvdSluijs said:
- phpstorm recommends you to add typehints
- laravel doesn't use them
- if add them on properties from parent classes, it breaks

Solution: don't add them for inherited properties.

**Comment by chrisan** [comment]
@DannyvdSluijs @mfn we may be talking past each other, let me rephrase one last time:

- I have no PHP inspects installed other than defaults.  If you set your PHP language level to 7.4 this warning/error from PhpStorm beings to appear.  If you set your PHP language level to anything below that these inspection warning/errors go away as 7.4 introduced typed properties (and there was much rejoicing)
- I don't have any plugins installed except for "Laravel 0.15.4 by Daniel Espendiller".  Disabling that plugin has no effect. 
- I know Laravel does not use them and Taylor does not want to add types (and now that he is on 8 there is no chance in hell 6.x ever gets touched for this)
- Adding a type hint array via a doc block does change anything
- I know laravel-ide-helper is not the cause of any of this
- My code runs fine without having them, the only "issue" is I do not get a green check mark for the file

So knowing all of that, my original intent of this issue was asking if there is anything that can be _added_  to laravel-ide-helper to help the ide when using php 7.4+ to make these weak warnings go away when using Laravel 6 or 7 (and likely 8 and beyond until Taylor gets on the typed program)

**Comment by mfn** [comment]
Thanks for the clarification!

**Comment by mfn** [comment]
Not sure how this could be solved, probably by generating stubs for PhpStorm (duplicate class warnings, hello?) for the Laravel framework? Anyone got a smarter idea?

----
My _strictly personal_ opinion:
- IMHO it's wrong for PhpStorm to suggest this for properties inherited from `vendor/` space which are inherently "don't modify"
  q.e.d, you can't even do that in fact:
  ![image](https://user-images.githubusercontent.com/87493/92406192-b1667a80-f137-11ea-9d4c-57deeaab3583.png)
  Seems I'm not the only one thinking this => https://youtrack.jetbrains.com/issue/WI-50847
- I don't see much gain to just figure out to work around a _minor_ weak warning annoyance in PhpStorm by adding more code = complexity to maintain. I.e. just don't see the "value gained".
  As much as I like to "hit F2 and don't see anything pointed out" in PhpStorm, some things may simply not worth it.
  In my experience I write 99% of the code _outside_ of the models so I rarely see any of those weak warnings to begin with.

**Comment by chrisan** [comment]
> In my experience I write 99% of the code outside of the models so I rarely see any of those weak warnings to begin with.

Ya same boat for me so if this is too much effort that is perfectly fine to skip.  The OCD green check mark monster made me open the issue, I can shove him in a closet :)

**Comment by mfn** [comment]
>  The OCD green check mark monster made me open the issue, I can shove him in a closet :)

I know that feeling :)

Thanks for the feedback, closing then!
